projects/components/src/utils/id-generator/id-generator.ts

```diff
@@ -0,0 +1,26 @@
+/*!
+ * Copyright 2019 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+/**
+ * Object containing the logic to generate unique identifiers whenever it's {@link IdGenerator#id} method is called
+ */
+export class IdGenerator {
+    /**
+     * Variable that is incremented whenever a new ID is needed
```

Comment:
No need for this comment. private variables should be documented if they are confusing.